pkg/library/flatten/flatten.go

```diff
@@ -127,6 +123,9 @@ func recursiveResolve(workspace *dw.DevWorkspaceTemplateSpec, tooling ResolverTo
 		}
 	}
 
+	if err := mergeVolumeComponents(resolvedContent, resolvedParent, pluginSpecContents...); err != nil {
```

Comment:
Maybe it makes sense to move merging logic to a separate file?

here we just call abstract merge, and inside merge volumes and merging with devfile/api.

Reply:
Good idea, I've moved this logic into `merge.go` with the more-abstract `mergeDevWorkspaceElements`